### ISSUE TITLE
Improve time to convert binning

### ISSUE THREAD

**Issue by paolodamico** [PostHog/posthog#5439]
## Bug description

While trying to understand the role of time in improving our session recording playback experience, I got the graph below ([reference](https://app.posthog.com/insights?insight=FUNNELS&interval=day&events=%5B%5D&actions=%5B%7B%22id%22%3A3091%2C%22name%22%3A%22KFA%20-%20Session%20recordings%22%2C%22type%22%3A%22actions%22%2C%22order%22%3A0%7D%2C%7B%22id%22%3A3091%2C%22name%22%3A%22KFA%20-%20Session%20recordings%22%2C%22type%22%3A%22actions%22%2C%22order%22%3A1%7D%2C%7B%22id%22%3A3091%2C%22name%22%3A%22KFA%20-%20Session%20recordings%22%2C%22type%22%3A%22actions%22%2C%22order%22%3A2%7D%2C%7B%22id%22%3A3091%2C%22name%22%3A%22KFA%20-%20Session%20recordings%22%2C%22type%22%3A%22actions%22%2C%22order%22%3A3%7D%2C%7B%22id%22%3A3091%2C%22name%22%3A%22KFA%20-%20Session%20recordings%22%2C%22type%22%3A%22actions%22%2C%22order%22%3A4%7D%2C%7B%22id%22%3A3091%2C%22name%22%3A%22KFA%20-%20Session%20recordings%22%2C%22type%22%3A%22actions%22%2C%22order%22%3A5%7D%5D&new_entity=%5B%5D&funnel_viz_type=time_to_convert&display=FunnelViz)). Problem with this graph is that the first comprises 90+% of the users, therefore this distribution graph tells me little about the actual shape of my time to convert distribution.  
  
![image](https://user-images.githubusercontent.com/5864173/128099593-0b8fcb24-6538-4644-9aa1-6e32e8aca27c.jpeg)
## Expected behavior

I would expect smarter binning where instead I would get a ton of detail into that first 90%+ and less detail into the remaining outliers.

## How to reproduce

1. Click on the graph provided above.

## Environment

- [X] PostHog Cloud
- [X] self-hosted PostHog, version/commit: _please provide_

#### *Thank you* for your bug report – we love squashing them!

**Comment by Twixes** [comment]
I would push back against variable bin widths. That would distort the viz and reduce its usefulness, as it's much easier to scan when you know that bin width corresponds to the underlying interval – without that it's not even a histogram anymore.

However what we could do is use significantly more bins. In the screenshot above we could easily show 20 or 30 bins instead of just 6, offering much better granularity while keeping the useful characteristics of a histogram. I checked a couple of our competitors' products and that's also what they resort to.

**Comment by paolodamico** [comment]
Something like that could make sense, maybe double clicking on a bin too. I'm not sure if the answer is just more bins, but if a single bin contains 90% of the data points we would definitely want more granularity here.

**Comment by clarkus** [comment]
> Something like that could make sense, maybe double clicking on a bin too. I'm not sure if the answer is just more bins, but if a single bin contains 90% of the data points we would definitely want more granularity here.

This is almost like a "drill-in" action to scope the entire histogram to the bounds of the selected bin, and then recalculating bins based on that new scope.

**Comment by paolodamico** [comment]
Indeed, I'm just not 100% sure removing these outliers and recalculating won't skew the data in a poor way. Any thoughts @Twixes ?

**Comment by neilkakkar** [comment]
Now that we can constrain the conversion window, another way out of the problem is to let the user themselves re-calculate time to convert by limiting the conversion window. Like here: https://github.com/PostHog/posthog/issues/5677#issuecomment-904649838

**Comment by neilkakkar** [comment]
I think that solution is good enough to close this issue?

For your reference chart:  
![image](https://user-images.githubusercontent.com/7115141/130627199-25c2bd13-d77c-46bc-b818-c8044ffa8aa7.png)


I switched the window to 1 day, since most things were happening within 8 hours.

Maybe just add good copy to explain this is possible?

**Comment by clarkus** [comment]
This looks like a good solution. We have a related issue at https://github.com/PostHog/posthog/issues/5557 that is targeted at keeping conversion windows aligned to the visualization time range. We could rely on users to keep this in sync, but I wonder if there's any basic business rules we could apply here to keep time options aligned. I haven't figured this out for the related issue either.

**Comment by luke-belton** [comment]
+1 [27453](https://posthoghelp.zendesk.com/agent/tickets/27453)
